### ISSUE TITLE
Add Traefik

### ISSUE THREAD

**Issue by Ornias1993** [jailmanager/jailman#3]
Currently encryption and proxy handling is a huge mess.
To cleanup, I propose adding a traefik proxy setup

**Comment by mikestecker** [comment]
Have you looked into [Traefik](https://containo.us/traefik/)? I currently use it as my reverse proxy on another docker machine and love it. I found your scripts when looking for how I'd migrate all my docker setup over to FreeNAS.

**Comment by Ornias1993** [comment]
I've heard of it, but this issue is one of the lowest priority ones, so I haven't yet looked much into practical application of it.

**Comment by thorsteneb** [comment]
Base differences between nginx and traefik. Traefik is very nice, agreed.

nginx: A high performance free open source web server powering busiest sites on the Internet. nginx [engine x] is an HTTP and reverse proxy server, as well as a mail proxy server, written by Igor Sysoev. According to Netcraft nginx served or proxied 30.46% of the top million busiest sites in Jan 2018; 

Traefik: Load Balancer for Microservices. Træfɪk is a modern HTTP reverse proxy and load balancer made to deploy microservices with ease. It supports several backends (Docker, Swarm, Mesos/Marathon, Kubernetes, Consul, Etcd, Zookeeper, BoltDB, Rest API, file...) to manage its configuration automatically and dynamically.

Good guide for nginx in a jail with Let's Encrypt and DDNS: https://fahadusman.com/setting-up-nginx-reverse-proxy-cloudflare-lets-encrypt-ssl-for-nextcloud-ddns-on-freenas-11-3-u1jail/

I did not find anything like that for Traefik. Traefik does exist on FreeBSD, however: https://github.com/containous/traefik/releases

**Comment by Ornias1993** [comment]
I don't know if you noticed, but a guide wasn't needed... The Organizr jail already runs a nginx reverse proxy... Thats why the issue is called "Seperate" not "Add reverse proxy jail".

It's in fact just meant to split the NGINX proxy inside organizr to something stand alone and set it up in such a way that other jails get autoconfigured.

Traefic wasn't my idea and honestly wasn't actually the scope of this issue to be frankfully honest.

**Comment by Ornias1993** [comment]
Okey, I looked into Traefik and actually experimented with it.
It's less documented than I would've liked, but I tried some Proof-of-Concepts already.

Some design notes:
- Traefik configuration via the static config (traefik.toml/traefik.yml), would be an "on traefik install"action
- Traefik can read all config files in a certain directory and update its config on changes in those files
- We can easily have install/remove/update scripts create or link files to said folder
- This would remove the requirement of including the (mediocre at best and not really supported) SSL certificate processing in Nextcloud
- We could easily add a "behind_organizr" tag for all jails, that would automatically put an forward_auth tag on them to require organizr auth before use.
